docs/source/en/optimization/torch2.0.md

```diff
@@ -78,6 +78,24 @@ For more information and different options about `torch.compile`, refer to the [
 > [!TIP]
 > Learn more about other ways PyTorch 2.0 can help optimize your model in the [Accelerate inference of text-to-image diffusion models](../tutorials/fast_diffusion) tutorial.
 
+### Regional compilation
+
+Compiling the whole model usually has a big problem space for optimization. Models are often composed of multiple repeated blocks. [Regional compilation](https://pytorch.org/tutorials/recipes/regional_compilation.html) compiles the repeated block first (a transformer encoder block, for example), so that the Torch compiler would re-use its cached/optimized generated code for the other blocks, reducing (often massively) the cold start compilation time observed on the first inference call.
+
+Enabling regional compilation might require simple yet intrusive changes to the
+modeling code. However, 🤗 Accelerate provides a utility [`compile_regions()`](https://huggingface.co/docs/accelerate/main/en/usage_guides/compilation#how-to-use-regional-compilation) which automatically _only_ compiles
+the repeated blocks of the provided `nn.Module`.
```

Comment:
no we actually compile the rest of the model as well 😅 I found out in my post that some people thought only the encoder/decoder block will be compiled in regional, which is not true.
I changed the docs to be more explicit https://github.com/huggingface/accelerate/pull/3572#discussion_r2089449059

Reply:
👁️ But https://docs.pytorch.org/tutorials/recipes/regional_compilation.html suggests a completely different recipe no? No full compilation but only regional and I always thought that is what should be done. 

What am I missing?

Reply:
regional compilation is simply: cut into regions and then compile them. I didn't compare the two approaches but I believe in the context of the pytorch tutorial they were simply trying to reduce cold start, not caring about how to keep inference optimized as well (they didn't benchamrk inference).

Reply:
So 

1. inference latency of compiling full model >= inference latency of regionally compiling repeated blocks + compiling additional blocks in a model
2. cold start time of compiling full model >> cold start time of regionally compiling repeated blocks + compiling additional blocks in a model

Is my understanding right or is it still fragmented?

Reply:
yes that is how it works !

> Do you think providing an option to NOT compile the rest of the blocks could still make sense?

doesn't sense for me personally, since you will miss on the tuning of the task-specific head. Do you have any specific cases where we don't want to compile the rest of the model ?